Detectors/TPC/simulation/include/TPCSimulation/DigitGlobalPad.h

```diff
@@ -123,8 +123,10 @@ inline void DigitGlobalPad::reset()
 
 inline bool DigitGlobalPad::compareMClabels(const MCCompLabel& label1, const MCCompLabel& label2) const
 {
-  return (label1.getEventID() == label2.getEventID() && label1.getTrackID() == label2.getTrackID() &&
-          label1.getSourceID() == label2.getSourceID());
+  // we compare directly on the bare label (in which eventID, labelID etc. are encoded)
+  // this avoids any logical operator on the label; optimization motivated from an Intel VTune analysis
+  // (note that this is also faster than using the operator= of MCCompLabel)
+  return label1.getRawLabel() == label2.getRawLabel();
```

Comment:
For clarification,
why cannot one simply use `return label1 == label2;`? Doesn't this do the trick? Or what is the difference to the raw label?

Reply:
it doesn't since this would call operator() of MCCompLabel which would mask out some weight information and would use bitshifts again.
(Since we don't care about the weight information, we can directly compare the raw un-masked label).

Reply:
Ok, there is the masking in `operator==`, but no bit shift. I still don't understand the comment about the weights. From MCCompLabel I see that the only relevant information are `int trackID, int evID, int srcID` which are set in `mLabel`.
From what I can judge, comparing the separate information (as done in the original code) and comparing the masked `mLabel` as done in the `operator==` is the same.
For me the logic of this commit is not clear. Either I can compare two labels using the `operator==` or not. If the masking is an issue of speed, why not comparing `mLabel` direclty in `operator==`.
If in the TPC code we use the raw labels `mLabel` for comparition, and at some point the usage of the `mLabel` field changes, we might run into problems.

Reply:
I didn't write this class. There are good reasons why in the general operator==() should just compare eventID, sourceID and trackID() and mask the remaining bits.
I am arguing that in the TPC case we don't care about the remaining bits so we can directly compare the raw label. This saves billions of masking operations and was shown to be beneficial by VTune.